### PR TITLE
Unify embedded RequestResponseDeploymentStrategy with K8DeploymentMan…

### PR DESCRIPTION
**Pull request by gadomsky**
…ager for RR

### CODE REVIEW

File: engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/requestresponse/RequestResponseDeploymentStrategy.scala [hunk 1]
```diff
@@ -80,23 +79,21 @@ class RequestResponseDeploymentStrategy(httpConfig: HttpBindingConfig, config: R
...
-      ProductionServiceInvocationCollector, ComponentUseCase.EngineRuntime)
-    val interpreterWithSlug = slugForScenario(jobData.metaData).product(interpreter)
-    interpreterWithSlug.foreach { case (slug, interpreter) =>
-      slugToScenarioRoute += (slug -> new ScenarioRoute(new RequestResponseAkkaHttpHandler(interpreter), config, jobData.processVersion.processName))
-      interpreter.open()
+    lazy val interpreterTry = Try(new RequestResponseRunnableScenarioInterpreter(jobData, parsedResolvedScenario, modelData = modelData, contextPreparer = contextPreparer, requestResponseConfig = config))
+
+    for {
+      slug <- slugForScenario(jobData.metaData).fold(errors => Failure(new IllegalArgumentException(errors.toString())), Success(_))
+      interpreter <- interpreterTry
```
Comment (mproch) [inline engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/requestresponse/RequestResponseDeploymentStrategy.scala:86]: where do we open interpreter here?

Comment (gadomsky) [reply engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/requestresponse/RequestResponseDeploymentStrategy.scala:86]: unfortunetly in constructor of `RequestResponseRunnableScenarioInterpreter` ...

File: engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/requestresponse/RequestResponseDeploymentStrategy.scala [hunk 2]
```diff
@@ -80,23 +79,22 @@ class RequestResponseDeploymentStrategy(httpConfig: HttpBindingConfig, config: R
...
-    import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter._
-
-    val interpreter = RequestResponseInterpreter[Future](parsedResolvedScenario, jobData.processVersion, contextPreparer, modelData, Nil,
-      ProductionServiceInvocationCollector, ComponentUseCase.EngineRuntime)
-    val interpreterWithSlug = slugForScenario(jobData.metaData).product(interpreter)
-    interpreterWithSlug.foreach { case (slug, interpreter) =>
-      slugToScenarioRoute += (slug -> new ScenarioRoute(new RequestResponseAkkaHttpHandler(interpreter), config, jobData.processVersion.processName))
-      interpreter.open()
+    // RequestResponseScenarioInterpreter is 'opened' in constructor of RequestResponseRunnableScenarioInterpreter
+    lazy val interpreterTry = Try(new RequestResponseRunnableScenarioInterpreter(jobData, parsedResolvedScenario, modelData = modelData, contextPreparer = contextPreparer, requestResponseConfig = config))
```
Comment (dswiecki) [inline engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/requestresponse/RequestResponseDeploymentStrategy.scala:83]: modeData and contextPreparer params have the same names as params in function declaration, maybe write just
new RequestResponseRunnableScenarioInterpreter(jobData, parsedResolvedScenario, modelData, contextPreparer, requestResponseConfig = config)

**Review by dswiecki — APPROVED** [review]
lgtm

**Pull request merged**